### PR TITLE
Improve consistency of struts with existing tests

### PR DESCRIPTION
**Pull request by trask**
cc: @vovencij

* add tests for redirect and error endpoints
* sets struts version to first 2.3 version in maven central (2.3.1)
* a couple other minor updates to be more consistent with other mvc framework tests

I had sent these updates as a [PR to #1628](https://github.com/vovencij/opentelemetry-java-instrumentation/pull/1), but I forgot to mention it in the PR, and then I accidentally deleted the branch which closed my PR 😅

### CODE REVIEW

**Comment by vovencij** [comment]
Hah, and I was wondering why did you just close that branch :)

File: instrumentation/struts-2.3/javaagent/src/test/java/io/opentelemetry/struts/GreetingAction.java [hunk 1]
```diff
@@ -20,25 +20,38 @@ public String success() {
...
+    HttpServerTest.controller(
+        HttpServerTest.ServerEndpoint.EXCEPTION,
+        () -> {
+          throw new Exception(HttpServerTest.ServerEndpoint.EXCEPTION.getBody());
+        });
+    throw new AssertionError(); // should not reach here
   }
 
-  public String pathParam() {
+  public String path_param() {
```
Comment (iNikem) [inline instrumentation/struts-2.3/javaagent/src/test/java/io/opentelemetry/struts/GreetingAction.java:54]: Why snake case?

Comment (vovencij) [reply instrumentation/struts-2.3/javaagent/src/test/java/io/opentelemetry/struts/GreetingAction.java:54]: I believe this is needed for `def expectedMethodName = endpoint.name().toLowerCase()` to work in the test.

Comment (trask) [reply instrumentation/struts-2.3/javaagent/src/test/java/io/opentelemetry/struts/GreetingAction.java:54]: ya, this "technique" is used in a few other tests as well, just staying consistent 😄

**Review by iNikem — APPROVED** [review]

**Pull request merged**